wrapanapi/lenovo.py

```diff
@@ -358,5 +360,84 @@ def inventory(self, *requested_items, **kwargs):
         requested_items = requested_items or self._inventory_available
         return {item: self._inventory_available[item](self, requester) for item in requested_items}
 
+    def _get_network_devices(self, server_name):
+        addin_cards = self._get_addin_cards(server_name) or []
+        pci_devices = self._get_pci_devices(server_name) or []
+        network_devices = []
+
+        for addin_card in addin_cards:
+            if (LenovoSystem._is_network_device(addin_card) and not
+                    LenovoSystem._is_device_in_list(addin_card, network_devices)):
+                network_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (LenovoSystem._is_network_device(pci_device) and not
+                    LenovoSystem._is_device_in_list(pci_device, network_devices)):
+                network_devices.append(pci_device)
+
+        return network_devices
+
+    def _get_storage_devices(self, server_name):
+        addin_cards = self._get_addin_cards(server_name) or []
+        pci_devices = self._get_pci_devices(server_name) or []
+        storage_devices = []
+
+        for addin_card in addin_cards:
+            if (LenovoSystem._is_storage_device(addin_card) and not
+                    LenovoSystem._is_device_in_list(addin_card, storage_devices)):
+                storage_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (LenovoSystem._is_storage_device(pci_device) and not
+                    LenovoSystem._is_device_in_list(pci_device, storage_devices)):
+                storage_devices.append(pci_device)
+
+        return storage_devices
+
+    @staticmethod
+    def _is_device_in_list(device, device_list):
+        device_id = LenovoSystem._get_device_unique_id(device)
+
+        for d in device_list:
+            if device_id == LenovoSystem._get_device_unique_id(d):
+                return True
+
+        return False
+
+    @staticmethod
+    def _is_network_device(device):
+        device_name = device.get("productName") or device.get("name")
+        device_name = device_name.lower()
+
+        return (device.get("class").lower() == "network controller" or
+                "nic" in device_name or
+                "ethernet" in device_name)
+
+    @staticmethod
+    def _is_storage_device(device):
+        device_name = device.get("productName") or device.get("name")
```

Comment:
docblock concerning the storage device classes that may be found, and why `serveraid` or `sd media raid` in the name is a safe comparison